code/modules/admin/topic.dm

```diff
@@ -2289,6 +2289,206 @@
+
+		input_text = P.parsepencode(input_text, admin_writer, usr) // Encode everything from pencode to html
+		qdel(admin_writer)
+
+		var/customname = input(src.owner, "Pick a title for the fax.", "Fax Title") as text|null
+		if(!customname)
+			customname = "paper"
+
+		var/sendername
+		switch(faxtype)
+			if("Central Command")
+				sendername = "Central Command"
+			if("Syndicate")
+				sendername = "UNKNOWN"
+			if("Custom")
+				sendername = input(owner, "What organization does the fax come from? This determines the prefix of the paper (i.e. Central Command- Title). This is optional.", "Organization") as text|null
+
+		if(sender)
+			notify = alert(owner, "Would you like to inform the original sender that a fax has arrived?","Notify Sender","Yes","No")
+
+		// Create the reply message
+		if(sendername)
+			P.name = "[sendername]- [customname]"
+		else
+			P.name = "[customname]"
+		P.info = input_text
+		P.update_icon()
+		P.x = rand(-2, 0)
+		P.y = rand(-1, 2)
+
+		if(destination != "All Departments")
+			if(fax.receivefax(P) == FALSE)
+				to_chat(owner, "<span class='warning'>Message transmission failed.</span>")
+				return
+		else
+			for(var/thing in GLOB.allfaxes)
+				var/obj/machinery/photocopier/faxmachine/F = thing
+				if(F.z in GLOB.station_z_levels)
+					addtimer(CALLBACK(src, .proc/handle_sendall, F, P), 0)
+
+		var/datum/fax/admin/A = new /datum/fax/admin()
+		A.name = P.name
+		A.from_department = faxtype
+		if(destination != "All Departments")
+			A.to_department = fax.department
+		else
+			A.to_department = "All Departments"
+		A.origin = "Custom"
+		A.message = P
+		A.reply_to = reply_to
+		A.sent_by = usr
+		A.sent_at = world.time
+
+		to_chat(src.owner, "<span class='notice'>Message transmitted successfully.</span>")
+		if(notify == "Yes")
+			var/mob/living/carbon/human/H = sender
+			if(istype(H) && H.stat == CONSCIOUS && (istype(H.ears, /obj/item/device/radio/headset)))
+				to_chat(sender, "Your headset pings, notifying you that a reply to your fax has arrived.")
+		if(sender)
+			log_admin("[key_name(src.owner)] replied to a fax message from [key_name(sender)]: [input_text]")
+			message_admins("[key_name_admin(src.owner)] replied to a fax message from [key_name_admin(sender)] (<a href='?_src_=holder;[HrefToken(TRUE)];AdminFaxView=[REF(P)]'>VIEW</a>).", 1)
+		else
+			log_admin("[key_name(src.owner)] sent a fax message to [destination]: [input_text]")
+			message_admins("[key_name_admin(src.owner)] sent a fax message to [destination] (<a href='?_src_=holder;[HrefToken(TRUE)];AdminFaxView=[REF(P)]'>VIEW</a>).", 1)
+		return
+
+	else if(href_list["refreshfaxpanel"])
+		if(!check_rights(R_FUN))
+			return
+
+		fax_panel(usr)
+		return
+
+	else if(href_list["EvilFax"])
+		if(!check_rights(R_FUN))
+			return
+		var/mob/living/carbon/human/H = locate(href_list["EvilFax"])
+		if(!istype(H))
+			to_chat(usr, "This can only be used on instances of type /mob/living/carbon/human")

```

Comment:
All chat sent to Goonchat needs to be marked up with span tags, this'll be really relevant in the future if we try to "dark mode" this window too.

Reply:
full stop at the end